### PR TITLE
Fix: Nav Item Api & Webhook and Functions stay selected

### LINKED ISSUE #7573: Settings Nav item doesn't stay selected
# Bug description

When clicking on a webhook, the `api & webhooks` menu item loose its selection

https://github.com/user-attachments/assets/c775cbf9-4e8c-4f3c-a43a-8ddcb4ee2523

# Desired Behavior

Should keep its selection like when navigating the data model settings

### PR DESCRIPTION
**Pull request by tushar110302**
Fixed Nav Item Api & Webhook and Functions stay selected like data model settings. 
 Now when clicked stays selected and deos not loose its selection

Fixes #7573

https://github.com/user-attachments/assets/4cb78158-8411-4ee1-9bcc-2870344c0c62

### CODE REVIEW

**Review by ehconitin — APPROVED** [review]
LGTM! @tushar110302 Thank you for contributing!

**Comment by tushar110302** [comment]
> LGTM! @tushar110302 Thank you for contributing!

Happy to contribute and become a part of it 🙂

**Comment by FelixMalfait** [comment]
Thanks @tushar110302! 
@ehconitin as discussed this works well but I think we should consider inverting the API so that people who add sub-pages to a page that didn't have any don't have to think about changing this setting. Currently the only page where we want to disable this is accounts where we display calendar/email directly

File: packages/twenty-front/src/modules/settings/components/SettingsNavigationDrawerItems.tsx [hunk 1]
```diff
@@ -109,7 +106,7 @@ export const SettingsNavigationDrawerItems = () => {
     return matchPath(
       {
         path: pathName,
-        end: !accountSubSetting.matchSubPages,
+        end: true,
       },
```
Comment (ehconitin) [inline packages/twenty-front/src/modules/settings/components/SettingsNavigationDrawerItems.tsx:110]: Hey @FelixMalfait! I'm thinking this change will really make the code more intentional, don't you agree? Or do you think we should revert it? I mean, it doesn't really make a difference unless we want matchSubPages as false for one of the Emails or Calendars, but I highly doubt that's the case. What do you think?

File: packages/twenty-front/src/modules/settings/components/SettingsNavigationDrawerItems.tsx [hunk 2]
```diff
@@ -90,14 +89,12 @@ export const SettingsNavigationDrawerItems = () => {
...
       path: SettingsPath.AccountsEmails,
       Icon: IconMail,
-      matchSubPages: true,
       indentationLevel: 2,
     },
     {
       label: 'Calendars',
       path: SettingsPath.AccountsCalendars,
       Icon: IconCalendarEvent,
-      matchSubPages: true,
```
Comment (ehconitin) [inline packages/twenty-front/src/modules/settings/components/SettingsNavigationDrawerItems.tsx:100]: same here

File: packages/twenty-front/src/modules/settings/components/SettingsNavigationDrawerItems.tsx [hunk 3]
```diff
@@ -90,14 +89,12 @@ export const SettingsNavigationDrawerItems = () => {
       label: 'Emails',
       path: SettingsPath.AccountsEmails,
       Icon: IconMail,
-      matchSubPages: true,
```
Comment (ehconitin) [inline packages/twenty-front/src/modules/settings/components/SettingsNavigationDrawerItems.tsx:93]: same

File: packages/twenty-front/src/modules/settings/components/SettingsNavigationDrawerItems.tsx [hunk 4]
```diff
@@ -43,7 +43,6 @@ type SettingsNavigationItem = {
   label: string;
   path: SettingsPath;
   Icon: IconComponent;
-  matchSubPages?: boolean;
```
Comment (ehconitin) [inline packages/twenty-front/src/modules/settings/components/SettingsNavigationDrawerItems.tsx:46]: same

File: packages/twenty-front/src/modules/settings/components/SettingsNavigationDrawerItems.tsx (hunk 1 above)
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/settings/components/SettingsNavigationDrawerItems.tsx:111]: I guess we can use the same pattern we used at the above level = default to true but leave it as an option that can be configured. Just did a quick edit

**Review by FelixMalfait — APPROVED** [review]

**Pull request merged**